addons/l10n_fr_hr_holidays/models/hr_leave.py

```diff
@@ -34,26 +32,28 @@ def _get_fr_date_from_to(self, date_from, date_to):
             raise UserError(_("An employee can't take paid time off in a period without any work hours."))
 
         if not self.request_unit_hours:
-            # extend the date range to the full day or requested period so that the difference of hours
-            # between employee's work hours and the company's work hours doesn't affect the computation
-            date_from = date_from.replace(hour=0, minute=0, second=0)
-            date_to = date_to.replace(hour=23, minute=59, second=59)
-
+            # Use company's working schedule hours for the leave to avoid duration calculation issues.
             def adjust_date_range(date_from, date_to, period, attendance_ids, employee_id):
-                period_ids = attendance_ids.filtered(lambda a: a.day_period == period)
-                max_hour = max(attendance.hour_to for attendance in period_ids)
-                min_hour = min(attendance.hour_from for attendance in period_ids)
-                date_from = self._to_utc(date_from, min_hour, employee_id)
-                date_to = self._to_utc(date_to, max_hour, employee_id)
+                period_ids_from = attendance_ids.filtered(lambda a: a.day_period in period
+                                                                    and int(a.dayofweek) == date_from.weekday()
+                                                                    and (not a.two_weeks_calendar or int(a.week_type) == a.get_week_type(date_from)))
+                period_ids_to = attendance_ids.filtered(lambda a: a.day_period in period
+                                                                    and int(a.dayofweek) == date_to.weekday()
+                                                                    and (not a.two_weeks_calendar or int(a.week_type) == a.get_week_type(date_to)))
```

Comment:
I don't think it's valid like this as a calendar that has been switched back to a normal 1 week calendar as the value isn't reset to False :thinking: You'll have to check `two_weeks_calendar` to be sure. This could also be a calendar method like `_get_attendances(target_date)`